projects/admin/src/app/app.component.ts

```diff
@@ -14,28 +14,50 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-import { Component } from '@angular/core';
+
+import { Component, OnInit, AfterViewChecked } from '@angular/core';
 import { UserService } from './service/user.service';
 import { User } from './class/User';
 import { AppConfigService } from './service/app-config.service';
+import { TranslateService } from '@ngx-translate/core';
 
 @Component({
   selector: 'admin-root',
   templateUrl: './app.component.html'
 })
-export class AppComponent {
+export class AppComponent implements OnInit {
 
   user: any;
 
+  access = false;
+
   constructor(
-    userService: UserService,
-    appSettings: AppConfigService
-    ) {
-    userService.getLoggedUser().subscribe(data => {
-      appSettings.setSettings(data.settings);
+    private userService: UserService,
+    private appSettings: AppConfigService,
+    private translate: TranslateService
+    ) { }
+
+  ngOnInit() {
+    this.userService.getLoggedUser().subscribe(data => {
+      // User not logged
+      if (!data.metadata) {
+        throw new Error('User is not logged');
+      }
+      const languages: Array<string> = this.appSettings.languages;
+      this.translate.addLangs(languages);
+      this.translate.setDefaultLang(this.appSettings.defaultLanguage);
+      this.translate.use(data.settings.language);
+      this.appSettings.setSettings(data.settings);
       const currentUser = new User(data.metadata);
-      console.log(currentUser, 'user', data);
-      userService.setCurrentUser(currentUser);
+      this.userService.setCurrentUser(currentUser);
```

Comment:
logic for loading and storing the user should be inside user.service.

Reply:
done